app/models/king.rb

```diff
@@ -13,6 +17,35 @@ def valid_move?(x,y)
     return false
   end
 
+
+  def can_castle?(x,y)
+    updated_at == created_at && \
+    ((x == 6 && castle_rook && castle_rook.x_position == 7 && \
+    castle_rook.updated_at == castle_rook.created_at && \
+    castle_rook.path_blocked?(5, y_position) == false) || \
+    (x == 2 && castle_rook && castle_rook.x_position == 0 && \
+    castle_rook.updated_at == castle_rook.created_at && \
+    castle_rook.path_blocked?(3, y_position) == false )) && \
+    [y, castle_rook.y_position].uniq == y_position
+  end
+
+
+  def castle_rook
+    @castle_rook = game.pieces.where(x_position: 0, type: Rook).first || game.pieces.where(x_position: 7, type: Rook).first
```

Comment:
This returns the first rook found, but not necessarily the correct rook as that is dependent on the castling direction.  This also returns *any* rook rather than only rooks of a friendly color.  Also, I believe the value of the `type` column will be a string - Rails may be smart enough to translate the class into a string, but it might be best to make it explicit:

```
def castle_rooks
  game.pieces.where(x_position: x_position, player_id: player_id, type: 'Rook')
end
```

That would return all rooks on the king's row that share the same player (color).  Combined with other checks inside `can_castle?`, you could pick which of those rooks (if any) are valid for castling.